core/src/consensus/tendermint/engine.rs

```diff
@@ -339,6 +346,23 @@ fn block_number_if_term_changed(
     Some(header.number())
 }
 
+fn inactive_validators(
```

Comment:
Does this function assume that every validator would get a chance to be a proposer at least once in a term?

Reply:
Yes, it is.

Reply:
The current term is 3600 seconds; there must be 360 blocks if no missing round.
Because the validator must have been delegated at least 1% of the total stakes, the validator must have a chance per 100 blocks in this worst case.